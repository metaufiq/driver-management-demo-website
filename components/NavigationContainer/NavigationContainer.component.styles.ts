import styled from 'styled-components'
import { 
  MenuItems as MenuItemsProps, 
  StyledComponentProps 
} from './NavigationContainer.component.types'

export const Navbar = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 3.5rem;
    margin: 0 2rem;
    background-color: white;
    @media (min-width: 768px) {
      justify-content: flex-end;
    }
`

export const InlineText = styled.p`
  display: inline-block;
`

export const UserName = styled(InlineText)`
  color: ${({theme})=>theme.colors.primary};
  margin-left: 0.3rem;
`

export const MenuIconOpen = styled.a<StyledComponentProps>`
    display: flex;
    justify-content: start;
    font-size: 1.5rem;
    color: ${({theme})=>theme.colors.primary};
    @media (min-width: 768px) {
      display: none;
    }
`

export const UserContainer = styled.div`
  align-items: center;
  justify-content: flex-end;
  display: flex;
  flex-direction: row;
`

export const MenuIconClose = styled.a`
    display: flex;
    justify-content: end;
    font-size: 0.8rem;
    margin-top: 0.75rem;
    margin-right: 1rem;
    color: ${({theme})=>theme.colors.primary};
    @media (min-width: 768px) {
      display: none;
    }
`

export const MenuList = styled.div<StyledComponentProps>`
    width: 250px;
    height: 100vh;
    background-color: white;
    @media (max-width: 768px) {
      width: 100%;
      max-height: ${({visible}) => visible ? 0 : '200px'};
      transition: all 0.3s ease-out;
    }
`

export const MenuItems = styled.li<MenuItemsProps>`
    list-style: none;
    display: flex;
    align-items: center;
    justify-content: start;
    width: 100%;
    padding: 1rem 0;
    border-left: ${({active,theme }) => active ? `4px solid ${theme.colors.primary}` : '0'};
`

export const MenuItemLinks = styled.a`
    display: flex;
    align-items: center;
    padding: 0 2rem;
    font-size: 0.875rem;
    font-weight: bold;
    text-decoration: none;
    color: gray;
    background-color: white;

    &:hover {
        color: ${({theme})=>theme.colors.primary};
    }
`

export const ContentContainer = styled.div`
  display: flex;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`

export const ChildContainer = styled.div`
  padding: 0;
  flex-grow: 1;
  background-color: #fafafa;
  padding: 2rem;
`
