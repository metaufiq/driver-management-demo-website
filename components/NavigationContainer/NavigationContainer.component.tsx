import { useState } from 'react'
import {FaBars} from 'react-icons/fa'

import { Component } from '../../index.types'
import { 
  ChildContainer,
  ContentContainer, 
  InlineText, 
  MenuIconClose, 
  MenuIconOpen, 
  MenuItemLinks, 
  MenuItems, 
  MenuList, 
  Navbar,
  UserContainer, 
  UserName 
} from './NavigationContainer.component.styles'
import { useRouter } from 'next/router'
import { Props } from './NavigationContainer.component.types'
import Avatar from '../Avatar'
import { NAVIGATION_MENU, USER_AVATAR_URL } from '../../constants'

const NavigationContainer: Component<Props> = (props) => {
    const router = useRouter()
    const isRootRoute = router.route === '/';
    
    const [close, setClose] = useState(false)
    
    const showSidebar = () => setClose(!close)
    return (
        <>
            <Navbar>
                <MenuIconOpen 
                  close={close} 
                  href="#" 
                  onClick={showSidebar}
                >
                    {close &&  <FaBars /> }
                </MenuIconOpen>
                <UserContainer>
                  <InlineText>Hello,</InlineText>
                  <UserName>Shipper User</UserName>
                  <Avatar 
                    src={USER_AVATAR_URL}
                    height={'30%'}
                    width={'30%'}
                  />
                </UserContainer>
            </Navbar>

            <ContentContainer>
              <MenuList close={close}>
                <MenuIconClose href="#" onClick={showSidebar}>
                    Close
                </MenuIconClose>
                  {NAVIGATION_MENU.map((item, index) => {
                      return (
                          <MenuItems 
                            key={index} 
                            active={
                              router.route === item.path ||
                              (isRootRoute && !!item.isRoot)
                            }>
                              <MenuItemLinks href={item.path}>
                                  {item.icon}
                                  <span style={{marginLeft: '16px'}}>{item.title}</span>
                              </MenuItemLinks>
                          </MenuItems>
                      )
                  })}
              </MenuList>
              <ChildContainer>
                {props.children}
              </ChildContainer>
            </ContentContainer>
        </>
    )
}

export default NavigationContainer