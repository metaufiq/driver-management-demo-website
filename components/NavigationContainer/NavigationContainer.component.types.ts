import { User } from "../../index.types"

export type Props = {
  children: JSX.Element,
}