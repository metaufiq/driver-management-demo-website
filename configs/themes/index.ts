const themes = {
  colors: {
    primary: '#ff453b'
  },
};

export { themes };