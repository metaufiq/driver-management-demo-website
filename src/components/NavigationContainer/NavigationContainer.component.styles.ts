import styled from 'styled-components'

export const ContentContainer = styled.div`
  display: flex;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`

export const ChildContainer = styled.div`
  padding: 0;
  flex-grow: 1;
  background-color: #fafafa;
  padding: 2rem;
`
