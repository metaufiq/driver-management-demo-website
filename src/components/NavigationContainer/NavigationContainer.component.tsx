import { ReactElement, useState } from 'react'
import { useRouter } from 'next/router'

import { Component } from '../../../index.types'
import { 
  ChildContainer,
  ContentContainer,
  MenuIconClose, 
  MenuItemLinks, 
  MenuItems, 
  MenuList,
} from './NavigationContainer.component.styles'
import { Props, SetVisible } from './NavigationContainer.component.types'
import { NAVIGATION_MENU, USER_AVATAR_URL } from '../../constants'
import Navbar from '../Navbar'

/**
 * function to set sidebar visibility
 * @param {SetVisible} setVisible set sidebar visible or not
 * @param visible visible condition for sidebar
 * @returns {VoidFunction} execute setVisible
 */
export const _showSidebar = (setVisible:SetVisible, visible: boolean): VoidFunction=>
() => setVisible(visible)

/**
 * Navigation Container components
 * @param {Props} props component props 
 * @returns {ReactElement} Text Input components
 */
const NavigationContainer: Component<Props> = (props: Props):ReactElement => {
    const router = useRouter()
    const isRootRoute = router.route === '/';
    
    const [visible, setVisible] = useState(false)
    
    return (
        <>
            <Navbar
              visible={visible}
              userImage={USER_AVATAR_URL}
              onPressBurger={_showSidebar(setVisible, true)}
            />

            <ContentContainer>
              <MenuList visible={visible}>
                <MenuIconClose href="#" onClick={_showSidebar(setVisible, false)}>
                    Close
                </MenuIconClose>
                  {NAVIGATION_MENU.map((item, index) => {
                      return (
                          <MenuItems 
                            key={index} 
                            active={
                              router.route === item.path ||
                              (isRootRoute && item.isRoot!)
                            }>
                              <MenuItemLinks href={item.path}>
                                  {item.icon}
                                  <span style={{marginLeft: '16px'}}>{item.title}</span>
                              </MenuItemLinks>
                          </MenuItems>
                      )
                  })}
              </MenuList>
              <ChildContainer>
                {props.children}
              </ChildContainer>
            </ContentContainer>
        </>
    )
}

export default NavigationContainer