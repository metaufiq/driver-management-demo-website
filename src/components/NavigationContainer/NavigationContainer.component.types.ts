import { SetState } from "../../../index.types"

export type SetVisible = SetState<boolean>

export type Props = {
  children: JSX.Element,
}