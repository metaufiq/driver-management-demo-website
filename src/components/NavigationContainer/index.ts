import NavigationContainer from "./NavigationContainer.component";

export default NavigationContainer