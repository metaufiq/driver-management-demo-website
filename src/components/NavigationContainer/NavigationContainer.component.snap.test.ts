import NavigationContainer from ".";
import { themes } from "../../configs/themes";
import { assertSnapshots } from "../../test";
import { Configs } from "../../test/index.types";

jest.mock('next/router', ()=>({
  ...jest.requireActual('next/router'),
  useRouter: ()=>({
    router:{
      route: '/'
    }
  })
}))

describe('NavigationContainerSnapshot', () => {
  const configs:Configs<any> = [
    {
      desc: 'should render navigation container',
      useTheme: true,
      theme: themes,
    }
  ];

  assertSnapshots(NavigationContainer, configs);
});
