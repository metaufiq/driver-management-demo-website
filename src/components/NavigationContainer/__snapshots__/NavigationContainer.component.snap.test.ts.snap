// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationContainerSnapshot should render navigation container 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi GyWKw"
  >
    <a
      class="sc-eCYdqJ kQaECE"
      href="#"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        />
      </svg>
    </a>
    <div
      class="sc-jSMfEi kJYybr"
    >
      <p
        class="sc-dkzDqf hMVRxV"
      >
        Hello,
      </p>
      <p
        class="sc-dkzDqf sc-hKMtZM hMVRxV idCBRc"
      >
        Shipper User
      </p>
      <img
        class="sc-bczRLJ hGUeOa"
        height="30%"
        src="https://randomuser.me/api/portraits/med/men/18.jpg"
        width="30%"
      />
    </div>
  </div>
  <div
    class="sc-jqUVSM kKkOBs"
  >
    <div
      class="sc-gKXOVf OnSFp"
    >
      <a
        class="sc-iBkjds euUWxU"
        href="#"
      >
        Close
      </a>
      <li
        class="sc-ftvSup iNSpeb"
      >
        <a
          class="sc-papXJ fCEdVy"
          href="#"
        >
          <svg
            baseProfile="tiny"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1.2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3s-6.186 5.34-9.643 8.232c-.203.184-.357.452-.357.768 0 .553.447 1 1 1h2v7c0 .553.447 1 1 1h3c.553 0 1-.448 1-1v-4h4v4c0 .552.447 1 1 1h3c.553 0 1-.447 1-1v-7h2c.553 0 1-.447 1-1 0-.316-.154-.584-.383-.768-3.433-2.892-9.617-8.232-9.617-8.232z"
            />
          </svg>
          <span
            style="margin-left: 16px;"
          >
            Beranda
          </span>
        </a>
      </li>
      <li
        class="sc-ftvSup iNSpeb"
      >
        <a
          class="sc-papXJ fCEdVy"
          href="/driver-management"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
            />
          </svg>
          <span
            style="margin-left: 16px;"
          >
            Driver Management
          </span>
        </a>
      </li>
      <li
        class="sc-ftvSup iNSpeb"
      >
        <a
          class="sc-papXJ fCEdVy"
          href="#"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M17 3h4a1 1 0 0 1 1 1v16a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h4V1h2v2h6V1h2v2zM4 9v10h16V9H4zm2 2h2v2H6v-2zm5 0h2v2h-2v-2zm5 0h2v2h-2v-2z"
              />
            </g>
          </svg>
          <span
            style="margin-left: 16px;"
          >
            Pickup
          </span>
        </a>
      </li>
    </div>
    <div
      class="sc-kDDrLX bcskTu"
    />
  </div>
</DocumentFragment>
`;
